planner/core/task.go

```diff
@@ -1467,14 +1471,50 @@ func (p *PhysicalHashAgg) attach2Task(tasks ...task) task {
 			attachPlan2Task(p, t)
 		}
 	} else if mpp, ok := t.(*mppTask); ok {
-		partialAgg, finalAgg := p.newPartialAggregate(kv.TiFlash)
-		if partialAgg != nil {
-			partialAgg.SetChildren(mpp.p)
-			mpp.p = partialAgg
+		// TODO : && t.count() > float64(p.SCtx().GetSessionVars().BroadcastJoinThresholdCount
+		if p.SCtx().GetSessionVars().AllowMPPExecution && p.tp == plancodec.TypeHashAgg {
```

Comment:
If we go to this branch, the condition is forever true, so this check is meaning less